compiler/fir/fir2ir/src/org/jetbrains/kotlin/fir/backend/generators/AnnotationGenerator.kt

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright 2010-2020 JetBrains s.r.o. and Kotlin Programming Language contributors.
+ * Use of this source code is governed by the Apache 2.0 license that can be found in the license/LICENSE.txt file.
+ */
+
+package org.jetbrains.kotlin.fir.backend.generators
+
+import org.jetbrains.kotlin.descriptors.annotations.AnnotationUseSiteTarget
+import org.jetbrains.kotlin.fir.FirAnnotationContainer
+import org.jetbrains.kotlin.fir.backend.Fir2IrVisitor
+import org.jetbrains.kotlin.fir.declarations.FirProperty
+import org.jetbrains.kotlin.fir.declarations.FirValueParameter
+import org.jetbrains.kotlin.ir.declarations.*
+import org.jetbrains.kotlin.ir.expressions.IrConstructorCall
+import org.jetbrains.kotlin.ir.util.isPropertyField
+import org.jetbrains.kotlin.ir.util.isSetter
+
+internal class AnnotationGenerator(private val visitor: Fir2IrVisitor) {
```

Comment:
Could you please provide some high-level comments/documentation? Doesn't have to be long. 🙂

Reply:
I cannot say that I personally need documentation here, but if Mark considers it to be necessary, please add it :)

Reply:
None of generators in the conversion module has such high-level comments, but that doesn't justify not adding one for this generator. :) This time, I'm doing my parts, including others that I added previously: https://github.com/JetBrains/kotlin/pull/3381/commits/8e45211aa5fc277ad01eeadd56b6b8816206386c